### PR TITLE
build.rs enhancements

### PR DESCRIPTION
**Pull request by king6cong**
1. Adapt android toolchain names
2. Use correct symbol prefix when using prebuilt jemalloc

### CODE REVIEW

File: jemalloc-sys/build.rs [hunk 1]
```diff
@@ -77,6 +77,21 @@ fn main() {
...
+            "Unprefixed `malloc` requested on unsupported platform `{}` => using prefixed `malloc`",
+            target
+        );
+        use_prefix = true;
+    }
+
+    if use_prefix {
+        println!("cargo:rustc-cfg=prefixed");
+    }
+
```
Comment (gnzlbg) [inline jemalloc-sys/build.rs]: What's the reason for this change?

File: jemalloc-sys/build.rs [hunk 2]
```diff
@@ -403,6 +406,10 @@ fn gnu_target(target: &str) -> String {
         "x86_64-pc-windows-msvc" => "x86_64-pc-win32".to_string(),
         "i686-pc-windows-gnu" => "i686-w64-mingw32".to_string(),
         "x86_64-pc-windows-gnu" => "x86_64-w64-mingw32".to_string(),
+        "armv7-linux-androideabi" => "arm-linux-androideabi".to_string(),
+        "aarch64-linux-android" => "aarch64-linux-android".to_string(),
+        "x86_64-linux-android" => "x86_64-linux-android".to_string(),
+        "i686-linux-android" => "i686-linux-android".to_string(),
```
Comment (gnzlbg) [inline jemalloc-sys/build.rs]: In the last three the name does not change so there is no need to add a match arm for them.

File: jemalloc-sys/build.rs (hunk 1 above)
Comment (king6cong) [reply jemalloc-sys/build.rs]: On systems where jemalloc must be prefixed:
1. the prebuilt jemalloc has to be prefixed
2. `link_name` should be adapt to this. So the `rustc-cfg=prefixed` should be emitted

File: jemalloc-sys/build.rs (hunk 2 above)
Comment (king6cong) [reply jemalloc-sys/build.rs]: Removed 😃

File: jemalloc-sys/build.rs (hunk 1 above)
Comment (gnzlbg) [reply jemalloc-sys/build.rs]: Sure, but IIUC this commit just moved the lines around, without any change to functionality.

Comment (king6cong) [reply jemalloc-sys/build.rs]: ```println!("cargo:rustc-cfg=prefixed");```

has to occur before the early ```return``` in L100 when ```JEMALLOC_OVERRIDE``` is set

File: jemalloc-sys/build.rs [hunk 3]
```diff
@@ -77,6 +77,21 @@ fn main() {
...
+    if !use_prefix && NO_UNPREFIXED_MALLOC.iter().any(|i| target.contains(i)) {
+        warning!(
+            "Unprefixed `malloc` requested on unsupported platform `{}` => using prefixed `malloc`",
+            target
+        );
+        use_prefix = true;
+    }
+
+    if use_prefix {
+        println!("cargo:rustc-cfg=prefixed");
```
Comment (king6cong) [inline jemalloc-sys/build.rs]: This has to occur before the early return in L100 when JEMALLOC_OVERRIDE is set

File: jemalloc-sys/build.rs (hunk 1 above)
Comment (gnzlbg) [reply jemalloc-sys/build.rs]: Ah indeed, thanks. So could you add a comment about this ? Otherwise this LGTM.

**Comment by gnzlbg** [comment]
Thanks!

File: jemalloc-sys/build.rs (hunk 1 above)
Comment (king6cong) [reply jemalloc-sys/build.rs]: Comment added 😃
